src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -123,7 +123,7 @@ public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
         public override Artifact VisitArtifact(Artifact node)
         {
             ArtifactLocation fileLocation = node.Location;
-            if (fileLocation != null && _run.OriginalUriBaseIds != null)
+            if (fileLocation != null && (_run.OriginalUriBaseIds != null || fileLocation.Uri != null))
```

Comment:
>(_run.OriginalUriBaseIds != null || fileLocation.Uri != nul [](start = 40, length = 59)

There's no need for the new check on `fileLocation.Uri`. In fact, there's no need for the existing check on `_run.OriginalUriBaseIds`. The check can be just:
```
if (fileLocation != null)
```
That's because the method `ArtifactLocation.TryReconstructAbsoluteUri` (see the call on Line 137 below) does those checks itself. It handles all the cases: whether `originalUriBaseIds` does or doesn't exist, and whether `file.Uri` is `null`, relative or absolute. And its parameter `out Uri uri` always has the right answer.

This means two things:

1. The bug was not that we were missing a check on `fileLocation.Uri` -- it was that the _existing_ check on `originalUriBaseIds` was wrong! The fix is just to remove that check. Because if `fileLocation.Uri` is absolute, it doesn't matter whether `originalUriBaseIds` is there or not. `TryReconstructAbsoluteUri` takes care of it all.

2. The two added lines 154 and 155 below are not necessary. Again, the parameter `out Uri uri`  is always the right answer.

Reply:
Nice fix. Thanks for the explanation.

---
In reply to: [316919117](https://github.com/microsoft/sarif-sdk/pull/1661#discussion_r316919117) [](ancestors = 316919117)